data-prepper-plugins/opensearch/src/test/java/org/opensearch/dataprepper/plugins/sink/opensearch/index/DynamicIndexManagerTests.java

```diff
@@ -0,0 +1,135 @@
+import org.mockito.ArgumentCaptor;
+import org.mockito.Mock;
+import org.opensearch.OpenSearchException;
+import org.opensearch.action.admin.cluster.settings.ClusterGetSettingsRequest;
+import org.opensearch.action.admin.cluster.settings.ClusterGetSettingsResponse;
+import org.opensearch.client.ClusterClient;
+import org.opensearch.client.IndicesClient;
+import org.opensearch.client.RequestOptions;
+import org.opensearch.client.ResponseException;
+import org.opensearch.client.RestClient;
+import org.opensearch.client.RestHighLevelClient;
+import org.opensearch.client.indices.CreateIndexRequest;
+import org.opensearch.client.indices.GetIndexRequest;
+import org.opensearch.client.indices.GetIndexTemplatesRequest;
+import org.opensearch.client.indices.GetIndexTemplatesResponse;
+import org.opensearch.client.indices.PutIndexTemplateRequest;
+import org.opensearch.dataprepper.model.event.JacksonEvent;
+import org.opensearch.dataprepper.plugins.sink.opensearch.OpenSearchSinkConfiguration;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.regex.Pattern;
+import java.time.format.DateTimeFormatter;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertThrows;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoMoreInteractions;
+import static org.mockito.Mockito.when;
+import static org.mockito.MockitoAnnotations.initMocks;
+
+public class DynamicIndexManagerTests {
+
+    private static final String ID = "id";
+    private static final String DYNAMIC = "dynamic";
+    private static final String INDEX_ALIAS = "test-${" + ID + "}-index-alias";
+    private static final String DATE_PATTERN = "yyyy.MM.dd";
+    private static final String INDEX_ALIAS_WITH_DATE_PATTERN = INDEX_ALIAS+ "-%{" + DATE_PATTERN + "}";
+
+    private IndexManagerFactory indexManagerFactory;
+    private IndexManager indexManager;
+
+    private DynamicIndexManager dynamicIndexManager;
+
+    @Mock
+    private RestHighLevelClient restHighLevelClient;
+
+    @Mock
+    private OpenSearchSinkConfiguration openSearchSinkConfiguration;
+
+    @Mock
+    private ClusterClient cluster;
+
+    @Mock
+    private ClusterGetSettingsResponse clusterGetSettingsResponse;
+
+    @Mock
+    private IndexConfiguration indexConfiguration;
+
+    @Mock
+    private IndicesClient indicesClient;
+
+    @Mock
+    private GetIndexTemplatesResponse getIndexTemplatesResponse;
+
+    @Mock
+    private RestClient restClient;
+
+    @Mock
+    private ResponseException responseException;
+
+    static final String EVENT_TYPE = "event";
+
+    @Before
+    public void setup() throws IOException {
+        initMocks(this);
+
+        indexManagerFactory = new IndexManagerFactory();
```

Comment:
This should be a mocked object. It should help you write the tests I suggested above.

Also, see my other comment about testing only one behavior at a time. Once you split the tests you won't need a real object here.